### ISSUE TITLE
[no-unused-vars] No error for unused class members

### ISSUE THREAD

**Issue by rhalaly** [typescript-eslint/typescript-eslint#965]
**Repro**

```JSON
{
  "rules": {
        "@typescript-eslint/no-unused-vars": [
            "error",
            {
                "vars": "all",
                "args": "after-used",
                "ignoreRestSiblings": false
            }
        ]
  }
}
```

```TS
class Foo {
   private bar: number= 0;
   private baz: boolean = false;

   public add() {
       this.bar++;
   }

   public decrease() {
       this.bar--;
   }

   private qux() {
       console.log('unused');
   }
}
```

**Expected Result**

Error for `baz` and `qux`.

**Actual Result**

No errors

**Additional Info**

It only catch unused vars inside the functions and args, but not for class members.

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `2.1.0` |
| `@typescript-eslint/parser`               | `2.1.0` |
| `TypeScript`                              | `3.5.3` |
| `ESLint`                                  | `6.3.0` |
| `node`                                    | `8.11.3` |
| `npm`                                     | `6.11.2` |
